templates/dns_zone.py

```diff
@@ -0,0 +1,82 @@
+""" This template creates a Cloud DNS zone for Private Google Access"""
+
+def generate_config(context):
+  """ Entry point for the deployment resources. """
+  project = context.properties['projectId']
+  zone_resource_name = context.properties['resourceName']
+
+  resources = []
+
+  zone_resource = {
+    'name': zone_resource_name,
+    # https://cloud.google.com/dns/docs/reference/v1/managedZones
+    'type': 'gcp-types/dns-v1:managedZones',
+    'properties': {
+      'description': 'Routes googleapis.com to restricted.googleapis.com VIP',
+      'dnsName': 'googleapis.com.',
+      'project': project,
+      'visibility': 'private',
+      'privateVisibilityConfig': {
+        'kind': 'dns#managedZonePrivateVisibilityConfig',
+        'networks': [{
+          'kind': 'dns#managedZonePrivateVisibilityConfigNetwork',
+          'networkUrl': context.properties['network']
+        }]
+      }
+    }
+  }
+
+  # If a dependsOn property was passed in, the network should depend on that.
+  if 'dependsOn' in context.properties:
+    zone_resource['metadata'] = {
+      'dependsOn': context.properties['dependsOn']
+    }
+  resources.append(zone_resource)
+
+
+  resources.append({
```

Comment:
It will be worth adding a comment helping a reader understand what this config is and how it works. Best reference is probably https://cloud.google.com/vpc-service-controls/docs/set-up-private-connectivity#configuring-dns

Reply:
definitely! thanks